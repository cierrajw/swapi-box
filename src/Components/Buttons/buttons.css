.buttons-section{
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 0 auto;
  width: 50rem;
  height: 15rem;
}

.card-section-button{
  border: 1px solid #106282;
  border: 2px solid #106282;
  height: 50px;
  width: 150px;
  background-color: black;
  color: gold;
  border-radius: 5px;
  font-size: 1.5em;
}

.card-section-button:hover{
  border: 1px solid #106282;
  border: 2px solid #106282;
  background-color: gold;
  color: #122148;
}


.button-selected{
  border: 2px solid #106282;
  height: 50px;
  width: 150px;
  background-color: gold;
  color: #122148;
  font-size: 1.5em;
  border-radius: 5px;
}
