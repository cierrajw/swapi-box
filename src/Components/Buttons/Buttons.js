import React, { Component } from 'react';
import './buttons.css'

class Buttons extends Component {
  constructor(){
    super();

    this.state={
      peopleButtonClicked: false,
      planetButtonClicked: false,
      vehiclesButtonClicked: false
    }
  }

  handlePeopleClicked = () =>{
    this.props.getPeopleCards();

    this.setState({
      peopleButtonClicked: true,
      planetButtonClicked: false,
      vehiclesButtonClicked: false
    })

  }

  handleVehicleClicked = () => {

    this.props.getVehicleCards();

      this.setState({
        vehiclesButtonClicked: true,
        peopleButtonClicked: false,
        planetButtonClicked: false
      })
  }

  render(){

    const { peopleButtonClicked, planetButtonClicked, vehiclesButtonClicked } = this.state;

    const isPeopleClicked = peopleButtonClicked ? "button-selected": "card-section-button";
    const isPlanetClicked = planetButtonClicked ? "button-selected": "card-section-button";
    const isVehiclesClicked = vehiclesButtonClicked ? "button-selected": "card-section-button";

    return(
      <section className="buttons-section">
      <button className={isPeopleClicked} onClick={this.handlePeopleClicked} name="people-button">People</button>
      <button className={isPlanetClicked} onClick={this.handleButtonClicked} name="planets-button">Planets</button>
      <button className={isVehiclesClicked} onClick={this.handleVehicleClicked} name="vehicles-button">Vehicles</button>
      </section>
    );
  }
}

export default Buttons;
