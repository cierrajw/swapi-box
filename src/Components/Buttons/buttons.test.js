import React from 'react';
import { shallow } from 'enzyme';
import Buttons from './Buttons.js';

describe('Buttons', () => {

  let mockDisplayCards;

  beforeEach(()=>{
    mockDisplayCards = jest.fn();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Buttons displayCards={mockDisplayCards}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
