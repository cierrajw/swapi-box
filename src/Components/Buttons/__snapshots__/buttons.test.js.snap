// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons
    displayCards={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="card-section-button"
          onClick={[Function]}
        >
          People
        </button>,
        <button
          className="card-section-button"
          onClick={[Function]}
        >
          Planets
        </button>,
        <button
          className="card-section-button"
          onClick={[Function]}
        >
          Vehicles
        </button>,
      ],
      "className": "buttons-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "card-section-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "card-section-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "card-section-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="card-section-button"
            onClick={[Function]}
          >
            People
          </button>,
          <button
            className="card-section-button"
            onClick={[Function]}
          >
            Planets
          </button>,
          <button
            className="card-section-button"
            onClick={[Function]}
          >
            Vehicles
          </button>,
        ],
        "className": "buttons-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "card-section-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "card-section-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "card-section-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
